### PR TITLE
feat: implement rich text editor, campaign analytics, donor management, image handling, and form enhancements

### PR DESCRIPTION
**Pull request by Lavanya-Alapana**
Rich Text Editor - WYSIWYG editor with formatting toolbar (no markdown needed)
Campaign Analytics - Total raised, donors, progress bars, recent donations
Donor Details - Complete donor info (name, email, phone) for admins
Image Management - Banner and product image uploads with preview
Character Limits - Description validation (100-160 characters)
Auto Slug Generation - Creates slug from title, prevents "new" conflicts
Product Labels - Clear labels for cost per unit, units required
Form Validation - Real-time validation and error handling
HTML Storage - Rich content storage with markdown fallback

### CODE REVIEW

File: src/lib/utils/image-url.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+): string {
+	if (!imagePath) {
+		return "";
+	}
+
+	if (imagePath.startsWith("http://") || imagePath.startsWith("https://")) {
+		return imagePath;
+	}
+
+	if (imagePath.startsWith("data:")) {
```
Comment (asiya-tabasum) [inline src/lib/utils/image-url.ts:13]: combine with above if

File: src/components/dashboard/campaigns/campaign-edit-form.tsx
Comment (asiya-tabasum) [inline src/components/dashboard/campaigns/campaign-edit-form.tsx:1]: file is not modular

File: src/components/dashboard/campaigns/rich-text-editor.tsx
Comment (asiya-tabasum) [inline src/components/dashboard/campaigns/rich-text-editor.tsx:1]: not modular

**Review by asiya-tabasum — CHANGES_REQUESTED** [review]

File: src/components/homepage/emergency-section.tsx [hunk 2]
```diff
@@ -12,6 +12,7 @@ import {
 	ChevronsLeft,
```
Comment (k34a) [inline src/components/homepage/emergency-section.tsx:12]: Why making changes to this file at all? If this is required at all, create a separate PR for this change

**Review by k34a — COMMENTED** [review]
Remove all changes which are unrelated to the admin dashboard - from this PR.
If they are required, create a separate branch/PR for them.

File: src/app/globals.css [hunk 3]
```diff
@@ -3,3 +3,161 @@
 @plugin "@tailwindcss/forms";
 @plugin "@tailwindcss/aspect-ratio";
 
+/* TipTap Editor Styles */
+.ProseMirror {
```
[the remark this answers is not in the document]
Comment (k34a) [reply src/app/globals.css:7]: Can you think of another way to keep these styles?
Currently globals.css is used everywhere in the application. Is there any way you can think of to keep these styles only in the pages that absolutely need these tiptap related styles?

[the remark this answers is not in the document]
Comment (k34a) [reply src/app/globals.css:7]: Yes, that's the way!

File: src/app/api/upload/route.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+		if (authResult.error) {
+			return NextResponse.json(
+				{ error: authResult.error },
+				{ status: authResult.status },
+			);
+		}
+
+		const formData = await request.formData();
+		const file = formData.get("file") as File;
+		const path = formData.get("path") as string;
```
[the remark this answers is not in the document]
Comment (Lavanya-Alapana) [reply src/app/api/upload/route.ts:36]: To avoid this, I’ll update the API to only accept a slug, and construct the path server-side as campaigns/${slug}/images

[the remark this answers is not in the document]
Comment (k34a) [reply src/app/api/upload/route.ts]: Read about presigned URL uploads

[the remark this answers is not in the document]
Comment (Lavanya-Alapana) [reply src/app/api/upload/route.ts]: should I go ahead and implement presigned URL uploads for the file upload API?

[the remark this answers is not in the document]
Comment (k34a) [reply src/app/api/upload/route.ts]: First read and understand this approach. Understand if it is better or worse than the current solution that you have implemented.

You should be able to explain your choice

[the remark this answers is not in the document]
Comment (Lavanya-Alapana) [reply src/app/api/upload/route.ts]: I’ve reviewed presigned URL uploads — they’re more secure and scalable since uploads go directly to Supabase, reducing API load. We control path and expiry too. I’d suggest switching for better performance and security, but if we prefer simpler logic for now, current API upload is still fine with size/traffic monitoring.

File: src/app/api/admin/campaigns/route.ts [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+		if (createError) {
+			console.error("Error creating campaign:", createError);
+			return NextResponse.json(
+				{ error: "Failed to create campaign" },
+				{ status: 500 },
+			);
+		}
+
+		// Create campaign products if provided
+		if (
```
[the remark this answers is not in the document]
Comment (Lavanya-Alapana) [reply src/app/api/admin/campaigns/route.ts:97]: I’ve added rollback logic. If product creation fails after creating the campaign, the campaign is automatically deleted to ensure consistency.

File: src/app/api/admin/campaigns/[slug]/route.ts [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+
+			if (productsToDelete.length > 0) {
+				await supabaseAdmin
+					.from("campaign_products")
+					.delete()
+					.in("id", productsToDelete);
+			}
+
+			// Update existing products
+			for (const product of newProducts) {
```
[the remark this answers is not in the document]
Comment (Lavanya-Alapana) [reply src/app/api/admin/campaigns/[slug]/route.ts:109]: i'll create a separate files for zod validation and product update logic

File: src/components/campaign/donormodal.tsx [hunk 7]
```diff
@@ -50,7 +50,7 @@ export default function DonorModal({
 		setDonors((prev) => [...prev, ...newDonors]);
 		setPage((p) => p + 1);
 		setLoading(false);
-	}, [campaignId]);
+	}, [campaignId, page]);
```
Comment (k34a) [inline src/components/campaign/donormodal.tsx:53]: Why add page as a dependency here? This function is setting the page state itself, so it will go into an infinite loop.
Think about it. Is there a better way to handle this scenario?

File: src/components/dashboard/campaigns/campaign-analytics.tsx [hunk 8]
```diff
@@ -0,0 +1,377 @@
...
+
+	const formatDate = (dateString: string) => {
+		return new Date(dateString).toLocaleDateString("en-IN", {
+			year: "numeric",
+			month: "short",
+			day: "numeric",
+		});
+	};
+
+	const calculateProgress = () => {
```
Comment (k34a) [inline src/components/dashboard/campaigns/campaign-analytics.tsx:82]: This function is independent of the component. Can be taken out

File: src/components/dashboard/campaigns/campaign-analytics.tsx [hunk 9]
```diff
@@ -0,0 +1,377 @@
...
+	const formatCurrency = (amount: number) => {
+		return new Intl.NumberFormat("en-IN", {
+			style: "currency",
+			currency: "INR",
+			minimumFractionDigits: 0,
+			maximumFractionDigits: 0,
+		}).format(amount);
+	};
+
+	const formatDate = (dateString: string) => {
```
Comment (k34a) [inline src/components/dashboard/campaigns/campaign-analytics.tsx:74]: This function is independent of the component. Can be taken out

File: src/components/dashboard/campaigns/campaign-analytics.tsx [hunk 10]
```diff
@@ -0,0 +1,377 @@
...
+			day: "numeric",
+		});
+	};
+
+	const calculateProgress = () => {
+		if (campaign.amount === 0) return 0;
+		return (campaign.collection / campaign.amount) * 100;
+	};
+
+	const getInitials = (name: string | null | undefined) => {
```
Comment (k34a) [inline src/components/dashboard/campaigns/campaign-analytics.tsx:87]: This function is independent of the component. Can be taken out

File: src/components/dashboard/campaigns/campaign-analytics.tsx [hunk 11]
```diff
@@ -0,0 +1,377 @@
...
+				setError("Failed to fetch analytics data");
+			} finally {
+				setLoading(false);
+			}
+		};
+
+		fetchAnalytics();
+	}, [campaign.slug]);
+
+	const formatCurrency = (amount: number) => {
```
Comment (k34a) [inline src/components/dashboard/campaigns/campaign-analytics.tsx:65]: This function is independent of the component. Can be taken out

File: src/components/homepage/categories/donation-card.tsx [hunk 12]
```diff
@@ -2,7 +2,7 @@
 import React from "react";
```
Comment (k34a) [inline src/components/homepage/categories/donation-card.tsx:2]: Changes in this file are not related to this PR. The sole purpose of this PR should be to give the admins a functionality to create and manage campaigns. If these are required, then make them in a separate branch and create a separate PR.

File: src/components/homepage/emergency-section.tsx (hunk 2 above)
Comment (k34a) [inline src/components/homepage/emergency-section.tsx:12]: Changes in this file are not related to this PR. The sole purpose of this PR should be to give the admins a functionality to create and manage campaigns. If these are required, then make them in a separate branch and create a separate PR.

File: src/app/api/admin/campaigns/route.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (Lavanya-Alapana) [reply src/app/api/admin/campaigns/route.ts:97]: i'll use a postgres function with supabase rpc function

[the remark this answers is not in the document]
Comment (Lavanya-Alapana) [reply src/app/api/admin/campaigns/route.ts:97]: so that campaign and its products are created together  
if anything fails the whole thing gets rolled back

[the remark this answers is not in the document]
Comment (k34a) [reply src/app/api/admin/campaigns/route.ts:97]: How about slug of the campaign?

[the remark this answers is not in the document]
Comment (k34a) [reply src/app/api/admin/campaigns/route.ts]: OK

[the remark this answers is not in the document]
Comment (k34a) [reply src/app/api/admin/campaigns/route.ts]: Test it once with all possible edge cases after creation
